### PR TITLE
`WPS350` enforce using augmented assign pattern

### LINKED ISSUE #789: Forbid incorrect assign pattern
# Rule request

## Thesis

We need to enforce:

```python
a += b
```

instead of:

```python
a = a + b
```

The same applies for all other operators: `-`, `&`, `**`, etc
https://rust-lang.github.io/rust-clippy/master/index.html#assign_op_pattern

## Reasoning

The first version is correct and short. This is a consistency rule.

### PR DESCRIPTION
**Pull request by slapshin**
-- Please check everything that applies: -->

- [x] I have double-checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [x] I have created at least one test case for the changes I have made
- [x] I have updated the documentation for the changes I have made
- [x] I have added my changes to the `CHANGELOG.md`

## Related issues
- Closes #789

### CODE REVIEW

File: tests/test_visitors/test_ast/test_statements/test_augmented_assign.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import pytest
+
+from wemake_python_styleguide.violations.consistency import (
+    AugmentedAssignPatternViolation,
+)
+from wemake_python_styleguide.visitors.ast.statements import (
+    AssignmentPatternsVisitor,
+)
+
+OPERATIONS = frozenset([
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_statements/test_augmented_assign.py]: `frozenset((`

We can make a new rule out of this! Do you want to create a request?

File: wemake_python_styleguide/violations/consistency.py [hunk 2]
```diff
@@ -1838,3 +1841,28 @@ class RedundantSubscriptViolation(ASTViolation):
...
+
+
+@final
+class AugmentedAssignPatternViolation(ASTViolation):
+    """
+    Enforce using augmented assign pattern.
+
+    Reasoning:
+        ``a += b`` is short and correct version of ``a = a + b``.
+        Why not use short version?
```
Comment (sobolevn) [inline wemake_python_styleguide/violations/consistency.py]: `Why not using the short version?`

File: wemake_python_styleguide/visitors/ast/statements.py [hunk 3]
```diff
@@ -427,3 +428,32 @@ def _has_wrong_keys(self, node: ast.keyword) -> bool:
...
+            len(node.targets) == 1 and
+            isinstance(node.value, ast.BinOp) and
+            isinstance(node.value.right, ast.Name) and
+            isinstance(node.value.left, ast.Name)
+        )
+
+        if not is_checkable:
+            return
+
+        bin_op = cast(ast.BinOp, node.value)
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/statements.py]: Can we please remove `cast` here?

File: tests/test_visitors/test_ast/test_statements/test_augmented_assign.py (hunk 1 above)
Comment (slapshin) [reply tests/test_visitors/test_ast/test_statements/test_augmented_assign.py]: Yes. We need a new rule) I'll create an issue. Also, have we rule about:
```
a in {1} # is not good, better: a == 1
```
?

Comment (sobolevn) [reply tests/test_visitors/test_ast/test_statements/test_augmented_assign.py]: Awesome! Let's do it too! 

Two new rules, that's what I like 👍

**Pull request merged**